src/placement/pl_map_common.c

```diff
@@ -365,30 +366,14 @@ determine_valid_spares(struct pool_target *spare_tgt, struct daos_obj_md *md,
 	if (spare_avail) {
 		/* The selected spare target is up and ready */
 		l_shard->po_target = spare_tgt->ta_comp.co_id;
-
-		/* XXX: Use pl_obj_shard::po_fseq to record the latest
-		 *      failure sequence of the targets on the remap
-		 *      chain for the given shard (@l_shard).
-		 *
-		 *      The f_shard->fs_fseq is the snapshot of the
-		 *      pool map version (that is incremental only)
-		 *      when related spare (or the original target)
-		 *      became down.
-		 *
-		 *      Currently, DAOS does not support the target
-		 *      re-integration. So the failure sequence for
-		 *      available spares will be the initial value
-		 *      (the oldest one). So here, we only need to
-		 *      consider those unavailable spares's failure
-		 *      sequences to find out the latest (largest).
-		 */
-		l_shard->po_fseq = f_shard->fs_fseq;
+		l_shard->po_fseq = spare_tgt->ta_comp.co_fseq;
```

Comment:
What's the purpose of this change? Isn't f_shared->fs_fseq == spare_tgt->ta_comp.co_fseq?

Reply:
No f_shard should be the shard we're replacing while spare_tgt is what we're replacing it with.
Without this change it appeared that sometimes when doing online drain/reintegration calculations 2 shards would end up with the same f_seq.

Reply:
I don't see why drain/reint needs this pl_obj_shard->po_fseq. It was used to store the "last fseq" of the shard, and DTX will use this information to choose leader.

The spare_tgt->ta_comp.co_fseq is zero in most cases, so this change will mistakenly store 0 instead of 'last fseq' in the 'po_fseq'.

BTW, the original comment is bit hard to understand, could you revise it? Thanks.